sosed/data_processing.py

```diff
@@ -107,6 +109,46 @@ def load_docword(self, ind: int) -> Dict[str, Counter]:
 
         return self._docword[ind]
 
+    def load_all_files_docword(self, ind: int) -> (Dict[str, Counter], str):
+        """
+        :param ind: index of docword file.
+        :return: mapping from repository names to counts of tokens in them.
+        """
+        doc_name = ""
+        if self._docword[ind] is None:
+            docword = {}
+
+            with self._docword_files[ind].open('r') as fin:
+                is_first = True
+                for line in fin:
+                    line = line.strip()
+                    if line:
+                        if is_first:
+                            doc_name = line
+                            is_first = False
+                            continue
```

Comment:
Instead of some complicated ifs, you can just read a single line first and proceed with iterating over the file.
```
doc_name = fin.readline()
for line in fin:
    ...
```